File: core/widget/src/commonMain/kotlin/now/shouldigooutside/core/widget/WidgetData.kt
package now.shouldigooutside.core.widget

import androidx.compose.runtime.Immutable
import kotlinx.serialization.Serializable
import now.shouldigooutside.core.model.score.ScoreResult

@Immutable
@Serializable
public data class WidgetData(
    val scoreResult: ScoreResult,
    val scoreLabel: String,
    val locationName: String,
    val formattedTemp: String,
    val formattedFeelsLike: String,
    val formattedWind: String,
    val precipChance: Int,
    val todayScoreResult: ScoreResult,
    val todayScoreLabel: String,
    val alertCount: Int,
    val updatedAtMillis: Long,
    val isStale: Boolean,
    val updatedAgoLabel: String,
    val activityName: String? = null,
)

File: apps/ios/SIGOWidget/WidgetData.swift
import Foundation

enum ScoreResult: String {
    case Yes, Maybe, No
}

struct WidgetData {
    let scoreResult: ScoreResult
    let scoreLabel: String
    let locationName: String
    let formattedTemp: String
    let formattedFeelsLike: String
    let formattedWind: String
    let precipChance: Int
    let todayScoreResult: ScoreResult
    let todayScoreLabel: String
    let alertCount: Int
    let updatedAtMillis: Int64
    let isStale: Bool
    let updatedAgoLabel: String
    let activityName: String?

    static let sample = WidgetData(
        scoreResult: .Yes,
        scoreLabel: "YES",
        locationName: "Toronto",
        formattedTemp: "22°C",
        formattedFeelsLike: "20°C",
        formattedWind: "12 km/h",
        precipChance: 10,
        todayScoreResult: .Yes,
        todayScoreLabel: "Yes",
        alertCount: 0,
        updatedAtMillis: Int64(Date().timeIntervalSince1970 * 1000),
        isStale: false,
        updatedAgoLabel: "",
        activityName: nil
    )
}
